### PR TITLE
Histogram smoketests

### PR DESCRIPTION
**Pull request by lyda**
Add smoke tests for histograms

This is a start on digitalocean/prometheus-client-c#6. Likely more
needed to test:
    
  * Each of the bucket functions.
  * Check the values of buckets and the sum.

### CODE REVIEW

**Comment by lyda** [comment]
Note this is based on the branch for #38.

File: promtest/test/promtest_histogram.c [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  for (int i = 0; i < PROMTEST_THREAD_POOL_SIZE; i++) {
+    if (*((int *)retvals[i]) != 0) {
+      TEST_FAIL_MESSAGE("thread did not exit properly");
+    }
+  }
+
+  // scrape the endpoint
+  FILE *f = popen("prom2json http://0.0.0.0:8000/metrics", "r");
+  if (f == NULL) {
+    TEST_FAIL_MESSAGE("shell out failed");
```
Comment (miroswan) [inline promtest/test/promtest_histogram.c:69]: Can you include the command that failed in the failure message?

File: promtest/test/promtest_histogram.c [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    TEST_FAIL_MESSAGE("Failed to extract array from JSON_Value");
+  }
+
+  for (int i = 0; i < json_array_get_count(collection); i++) {
+    JSON_Object *obj = json_array_get_object(collection, i);
+    if (obj == NULL) {
+      TEST_FAIL_MESSAGE("failed to retrieve single object within JSON_Array");
+    }
+    const char *name = json_object_get_string(obj, "name");
+    if (strcmp(name, "foo_histogram")) {
```
Comment (miroswan) [inline promtest/test/promtest_histogram.c:164]: What happens if -1 is returned from strcmp?

File: promtest/test/promtest_histogram.c [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    }
+    if (json_array_get_count(samples) < 1) {
+      TEST_FAIL_MESSAGE("No samples found");
+    }
+    JSON_Object *sample = json_array_get_object(samples, 0);
+    if (sample == NULL) {
+      TEST_FAIL_MESSAGE("failed to get metric sample");
+    }
+    *value = (char *)json_object_get_string(sample, "count");
+    break;
```
Comment (miroswan) [inline promtest/test/promtest_histogram.c:179]: Can you add a comment explaining the intention for the break statement here?

**Comment by miroswan** [comment]
@lyda Thank you for this contribution.

**Review by miroswan — APPROVED** [review]

**Pull request merged**